src/paradigma/heart_rate/feature_extraction.py

```diff
@@ -0,0 +1,288 @@
+    statistic: str
+        The statistic to be computed ['mean', 'var', 'median', 'kurtosis', 'skewness']
+        
+    Returns
+    -------
+    list
+        the statistic for the sensor segments
+    """
+    if statistic == 'mean':
+        return [np.mean(np.abs(x)) for x in sensor_col]
+    elif statistic == 'var':
+        return [np.var(x, ddof=1) for x in sensor_col]  # ddof=1 for unbiased variance is used, same as matlab
+    elif statistic == 'median':
+        return [np.median(np.abs(x)) for x in sensor_col]
+    elif statistic == 'kurtosis':
+        return [kurtosis(x, fisher = False) for x in sensor_col] # Fisher's definition of kurtosis is used, same as matlab
+    elif statistic == 'skewness':
+        return [skew(x) for x in sensor_col]
+    else:
+        raise ValueError("Statistic not recognized.")
+
+def compute_signal_to_noise_ratio(
+        ppg_segments: np.ndarray
+    ) -> list:
+    """
+    Compute the signal to noise ratio of the PPG signal.
+    
+    Args:
+    arr_ppg (numpy.ndarray): PPG signal.
+    
+    Returns:
+    list: Signal to noise ratio of the PPG windows.
+    """
+    l_signal_to_noise_ratios = []
+    for segment in ppg_segments:
+        arr_signal = np.var(segment)
+        arr_noise = np.var(np.abs(segment))
+        signal_to_noise_ratio = arr_signal / arr_noise
+        l_signal_to_noise_ratios.append(signal_to_noise_ratio)
+    
+    return l_signal_to_noise_ratios
+
+def compute_auto_correlation(
+        ppg_segments: np.ndarray, 
+        fs: int
+    ) -> list:
+    """
+    Compute the autocorrelation of the PPG signal.
+    
+    Args:
+        ppg_signal (np.ndarray): 2D array where each row is a segment of the PPG signal.
+        fs (int): Sampling frequency of the PPG signal.
+    
+    
+    Returns:
+        list: Autocorrelation of the PPG segments.
+    """
+    l_auto_correlations = []
+
+    for segment in ppg_segments:
+        autocorrelations = biased_autocorrelation(segment, fs*3)
+        peaks, _ = find_peaks(autocorrelations, height=0.01)
+        peak_values = autocorrelations[peaks]
+        sorted_peaks = np.sort(peak_values)[::-1]
+        if len(sorted_peaks) > 0:
+            auto_corr = sorted_peaks[0]
+        else:
+            auto_corr = 0
+        l_auto_correlations.append(auto_corr)
+
+    return l_auto_correlations
+
+def biased_autocorrelation(
+        x: np.ndarray, 
+        max_lag: int
+    ) -> np.ndarray:
+    """
+    Compute the biased autocorrelation of a signal (similar to matlabs autocorr function), where the normalization factor 
+    is the length of the original signal, and boundary effects are considered.
+    
+    Args:
+        x (np.ndarray): Input signal (1D array).
+        max_lag (int): Maximum lag to compute autocorrelation.
+    
+    Returns:
+        np.ndarray: Biased autocorrelation values for lags 0 to max_lag.
+    """
+    x = np.array(x) # Ensure x is a numpy array instead of a list
```

Comment:
Ik weet niet precies wat je wil, maar je zou ook gebruik kunnen maken van `np.asarray`: https://stackoverflow.com/questions/14415741/what-is-the-difference-between-np-array-and-np-asarray

Reply:
eens, maar zie nu eig ook dat hij overbodig is want de input is sws al een np.array